core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -47,6 +47,14 @@ extern bool show_warnings() noexcept;
 
 namespace Impl {
 
+// Cuda Teams use (team_size + 2)*sizeof(double) shared memory for team
+// reductions. They also use one int64_t in static shared memory for a shared
+// ID. Furthermore, they use additional scratch memory in some reduction
+// scenarios, which depend on the size of the value_type and is NOT captured
+// here.
+constexpr size_t cuda_team_max_reserved_shared_mem =
```

Comment:
Why is there `max` in this name? It's a minimum reservation, isn't it?

Reply:
no, its the max you might get because the actual calculation is: `(team_size+2)*sizeof(double)+sizeof(int64_t)`

Reply:
and max team size if 1024

Reply:
Ah, ok. Is there a `int max_team_size = 1024` somewhere? If not, could you make one, and use that by name, rather than the magic constant that I had to ask about?

Reply:
there isn't a constexpr one, and technically that might not be correct.  In some ways that number is more prone to misinterpretation or misuse than the `reserved_shared_mem`. Essentially, in most cases the max team size will be smaller than that due to stuff like register constraints. Misuse or misunderstanding what a "max_team_size" variable means is very likely to be much more harmful than misunderstanding what cuda_team_max_reserved_shared_mem means. The latter would simply mean you are underestimating a bit how much shared memory you could theoretically use per block, the max_team_size misuse could mean all kinds of problems since it will almost always be wrong.

Reply:
Even if it were just a variable local to this function, it would be an improvement. Perhaps `int max_possible_team_size`? `int theoretical_max_team_size`? `int team_size_upper_bound`?

Reply:
done